doc/apiary/v2/fiware-ngsi-simple.apib

```diff
@@ -50,16 +50,136 @@ to keep your client decoupled from implementation details.
 
 ## Context Entities [/v2/entities{?type,pattern,limit,offset}]
 
-### List entities [GET /v2/entities{?limit,offset,count,type,id}]
-
-Detailed description
-
-+ Parameters
-    + limit: 20 (optional, number) - Limit the number of entities to be retrieved
-    + offset: 20 (optional, number) - Skip a number of records. Pagination
-    + type: Room (optional, string) - Only retrieve entities of this type
-    + id: Bcn-Welt (optional, string) - Retrieve entities by id
-    + idPattern: (optional, string) - Retrieve entities by pattern id
+### List entities [GET /v2/entities{?limit,offset,count,type,id,idPattern,q,geometry,coords,attrs}]
+
+Retrieves a list of entities, according to a set of matching criteria defined by the `id`, `type`, `idPattern`, `q`,
+`geometry` and `coords` attribute (see below for a detailed description of these parameters). A given entity
+have to match all the criteria to be retrieved (i.e. criteria are combined in a logical AND way).
+
+The response payload is a vector, which elements are JSON objects corresponding to an entity, as follows:
+
+* `id` and `type` are the entity ID and entity type, respectively. In the case the entity has an attribute named `id` and/or `type` then they are ignored.
+* For entities without type (i.e. type equals to "" internally at CB), the `type` is not included
+* All other first level field are the attributes of the entity. The value of the field may use one of the following styles:
+ * If the attribute doesn't have type (i.e. type equal to "" internally at CB) or the type is "number" and at the same time the attribute doesn't have metadata, then the value of the field is directly the value of the attribute. If the type is "number", then the value doesn't use "", e.g: `{ "temperature": 35.4}`. Note that in some cases, the value of the attribute could be a JSON vector or a JSON object.
+ * In other cases, the value of the field is a JSON object with the following subfields: `value` (for the attribute value), `type` (for the attribute type, omitted if the attribute doesn't have a type) and maybe other fields (representing metadata).
+* Regarding fields representing attribute metadata, they are included within the JSON element representing the attribute, according to the following (very similar to the way an attribute is represented at first level):
+ * Metadata with name `value` or `type` are ignored (they collide with attribute value and type)
+ * If the metadata doesn't have type (i.e. type equal to "" internally at CB) or the type is "number"  then the value of the field is directly the value of the metadata. If the type is "number", then the value doesn't use "", e.g: `{ "accuracy": 2}`.
+ * In other cases, the value of the field is a JSON object with the following subfields: `value` (for the metadata value) and `type` (for the metadata type).
+
```

Comment:
In order to make it shorter, the description of the JSON for entities (which is common to many operations) could be moved to a separated section, including here just one reference to it (similar idea to the one you propose for the error payload).